src/plugin-api/hooks/hook.js

```diff
@@ -73,27 +73,43 @@ export default class Hook {
 	}
 
 	/**
-	 * The ordered raw values
+	 * The ordered hook values
 	 *
-	 * @type {Array<RawHookValue>}
+	 * @type {Array<HookValue>}
 	 */
-	get orderedRawValues() {
-		let orderedValues = [];
-
-		for (const rawValue of this.rawValues) {
-			const { value, name, before, after } = rawValue;
-			const orderedValue = { value, name };
-			const insertionTarget = before || after;
-			if (!insertionTarget) {
-				orderedValues.push({ value, name });
-				continue;
-			}
+	get orderedValues() {
+		// all hooks without before/after can stay in the order as they were added
+		const orderedValues = this.rawValues
+			.filter(v => !(v.before || v.after))
+			.map(({ name, value }) => ({ name, value }));
+		const needOrdering = this.rawValues.filter(v => v.before || v.after);
+		if (!needOrdering) {
```

Comment:
Filter returns an empty array if so check the length
```suggestion
		if (!needOrdering.length) {
```